lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -160,7 +160,10 @@ public void write(Page page)
 
         while (page != null) {
             int chunkRows = min(page.getPositionCount(), writerOption.getBatchSize());
-            Page chunk = page.getRegion(0, chunkRows);
+            Page chunk = page;
+            if (chunkRows < page.getPositionCount()) {
```

Comment:
ya https://github.com/trinodb/trino/pull/17385

Reply:
I don't know, most of the existing usages of `page.getRegion(0, positionsCount)` were already doing this check themselves, so got double checking from that PR for all the other usages anyway.

Reply:
> so we got double checking from that PR

every such place should be revisited in that PR, because when else is it going to be updated?

it's just around 20 places, so not a big deal

(i actually did the work, but didn't other places where the check is duplicated)